### ISSUE TITLE
Ram doesn't go above 2gb

### ISSUE THREAD

**Issue by nedellis** [WesCook/Homestead#28]
Game doesn't go above 2gb, this is causing huge lag spikes.  
I have tested Resonant rise because it's in AtLauncher too, and it reach 2.6gb with no problems.  
I also play FTB Infinity and it runs fine with no problem on ram usage.  
Java version is latest 8v144.

**Comment by WesCook** [comment]
I'm not able to replicate this.  Java/Minecraft allows me to allocate 4GB without problem.

![2017-08-16 16_44_30-](https://user-images.githubusercontent.com/792101/29389975-3399ad72-82a2-11e7-9167-3fafd737c588.png)

I would verify that Java x64 is being loaded by the ATLauncher, as determined by the path set in the launcher.  Your Windows will also need to be x64 bit, else it can't allocate larger amounts of ram.

I'm afraid this isn't to do with the pack itself, but your system's software/configuration instead.  So I'm going to close this issue for now.

**Comment by nedellis** [comment]
I have same situation like you, the fact is that mem used won't go over 2000. 

For example in your screenshot it says: mem: 1531/4062. On my pc mem used never go above 2000-2100ish/4062.  So it's like 2gb are allocated but not used.  
So every time it reaches 2000 ~ 2100 it lags, like it tries to empty ram. It goes around 1500 then it fills again at 2000 ~ 2100 and lags again, over and over. 
In the Jungle biome it's totally unplayable because of this mem lag. Because mem stays at 2100/4062 and keep lagging almost every sec.

On same pc FTB infinity/beyond or resonant rise on AtLauncher i often reach 3200/4062 with no problems. 

Hope it's clear now, sorry for my english.

I will try to format as soon as i can, but it's strange this behaviour happens only with your pack and not with even heavier pack.

**Comment by nedellis** [comment]
![javaw 2017-08-17 13-09-25-61](https://user-images.githubusercontent.com/12676613/29409640-82544a98-834d-11e7-829e-26b30c771ad6.png)

After few minutes of play ram goes up to 2000ish and lag starts.

**Comment by nedellis** [comment]
Tested the pack on a friend of mine pc (similar spec to mine) freshly formatted. 
Same issue. Definitely it's the pack. My friend is experienced programmer, he said it's something about java garbage collector. 
As i said, it's like the game tries to clear ram too aggressively with no reason at 2000mb occupied even if there is still 2000mb available ram (as you can see in the screenshot above).
It must be some mod or a config in this pack. 
Tomorrow i'll try to run the pack by removing mods and see what happened.

**Comment by nedellis** [comment]
Tried the modpack on a third (even more powerful) machine...Problem still occurs. 
The game ignores 2gb allocated memory, it tries to aggresively empty ram when the game hits 2000ish memory occupied. 
At this point it's clear that there is a big memory issue in this pack.

I tried to removed half the mods but mem used dropped to 500/800 so i'm unable to replicate the issue this way. 
Yet even with lesser mods installed feels strange that mem used is so low. Even your screenshot proves me that there is a problem. 1500mb with 147 mods it's not typical behaviour for a modpack like this. 
I'm veteran modded mc player and this is the first time i see this memory behave. 
Maybe there's a mod that alters memory usage, i don't know man.
 
Sorry but i give up at this point, i'm running out of solutions. Maybe the only way is to rebuild the pack from scratch and test it multiple times.  
So sad, this pack is probably the best survival experience i've ever had in this game. But it's unplayable with this situation for me. 
Probably the issue is not well know because lag occurs mainly in biomes with lot of trees, player population is low and most of the times people don't even see or ignore lag like this. 
:(

Very good job for the effort. Pack is still great.
